internal/ext/exporter.go

```diff
@@ -132,6 +132,7 @@ func (e *Exporter) Export(ctx context.Context, encoding Encoding, w io.Writer) e
 					Type:        f.Type.String(),
 					Description: f.Description,
 					Enabled:     f.Enabled,
+					Metadata:    f.Metadata,
```

Comment:
@kvnhmn the issue with CI looks like it stems from here. `structpb.Struct` marshalls as json using the following code: https://github.com/protocolbuffers/protobuf-go/blob/c33baa8f3a0d35fd5a39e43c22a50a050f707d34/types/known/structpb/struct.pb.go#L226-L228

This looks to me like it will always output `Fields` as a key since it is exported https://github.com/protocolbuffers/protobuf-go/blob/v1.34.2/types/known/structpb/struct.pb.go#L194

The CLI export tests as well as the general export tests are failing because the data thats being added here (metadata) is marshalling as:

```
metadata:
  fields: {}
```

https://github.com/flipt-io/flipt/actions/runs/9863131371/job/27235210359#step:3:623

Which the current tests do not expect.

I think we likely want to fix this by doing the following:

- implementing our own `MarshalJSON` method for the Metadata type (by likely wrapping it in a new type) and check if its empty, and do not marshal if its empty.
- Also we likely want to call the [AsMap](https://github.com/protocolbuffers/protobuf-go/blob/c33baa8f3a0d35fd5a39e43c22a50a050f707d34/types/known/structpb/struct.pb.go#L217-L224) method on it if non-nil, which will prevent the `fields` key from being marshalled and return a plain map

Reply:
cc @erka @GeorgeMac

Reply:
I added an example in a review comment below. I don't think we need a custom marshal json method. I think we just need, as suggested, to make `Metadata` on the `ext/common.go` version of flag to be a `map[string]string`. Then use `AsMap` on export. There is an similar method for adapting a map into a structpb on import too.

Then we should be OK to just let `omitempty` do its thing.